ui/server/src/main/scala/pl/touk/nussknacker/ui/service/ProcessToolbarService.scala

```diff
@@ -0,0 +1,119 @@
+    ProcessToolbarSettings.fromConfig(toolbarConfig, process)
+  }
+}
+
+object ProcessToolbarSettings {
+
+  import ToolbarHelper._
+
+  def fromConfig(processToolbarConfig: ProcessToolbarsConfig, process: BaseProcessDetails[_]): ProcessToolbarSettings =
+    ProcessToolbarSettings(
+      createProcessToolbarUUID(processToolbarConfig.uniqueCode, process),
+      processToolbarConfig.topLeft.filterNot(tp => verifyCondition(tp.hide, process)).map(tp => ToolbarPanel.fromConfig(tp, process)),
+      processToolbarConfig.bottomLeft.filterNot(tp => verifyCondition(tp.hide, process)).map(tp => ToolbarPanel.fromConfig(tp, process)),
+      processToolbarConfig.topRight.filterNot(tp => verifyCondition(tp.hide, process)).map(tp => ToolbarPanel.fromConfig(tp, process)),
+      processToolbarConfig.bottomRight.filterNot(tp => verifyCondition(tp.hide, process)).map(tp => ToolbarPanel.fromConfig(tp, process))
+    )
+}
+
+@JsonCodec
+case class ProcessToolbarSettings(uuid: UUID, topLeft: List[ToolbarPanel], bottomLeft: List[ToolbarPanel], topRight: List[ToolbarPanel], bottomRight: List[ToolbarPanel])
+
+object ToolbarPanel {
+
+  import ToolbarHelper._
+
+  def apply(`type`: ToolbarPanelType, title: Option[String], buttonsVariant: Option[ToolbarButtonVariant], buttons: Option[List[ToolbarButton]]): ToolbarPanel =
+    ToolbarPanel(`type`.toString, title, buttonsVariant, buttons)
+
+  def fromConfig(config: ToolbarPanelConfig, process: BaseProcessDetails[_]): ToolbarPanel =
+    ToolbarPanel(
+      config.identity,
+      config.title.map(t => fillByProcessData(t, process)),
+      config.buttonsVariant,
+      config.buttons.map(buttons =>
+        buttons
+          .filterNot(button => {
+            verifyCondition(button.hide, process)
+          })
+          .map(button => ToolbarButton.fromConfig(button, process))
+      )
+    )
+}
+
+@JsonCodec
+case class ToolbarPanel(id: String, title: Option[String], buttonsVariant: Option[ToolbarButtonVariant], buttons: Option[List[ToolbarButton]])
+
+object ToolbarButton {
+
+  import ToolbarHelper._
+
+  def fromConfig(config: ToolbarButtonConfig, process: BaseProcessDetails[_]): ToolbarButton = ToolbarButton(
+    config.`type`,
+    config.title.map(t => fillByProcessData(t, process)),
+    config.icon.map(i => fillByProcessData(i, process)),
+    config.templateHref.map(th => fillByProcessData(th, process)),
+    disabled = verifyCondition(config.disabled, process)
+  )
+}
+
+@JsonCodec
+case class ToolbarButton(`type`: ToolbarButtonType, title: Option[String], icon: Option[String], url: Option[String], disabled: Boolean)
+
+private [service] object ToolbarHelper {
+
+  def createProcessToolbarUUID(baseCode: String, process: BaseProcessDetails[_]): UUID =
+    UUID.nameUUIDFromBytes(s"$baseCode-${process.isSubprocess}-${process.isArchived}".getBytes())
+
+  def fillByProcessData(text: String, process: BaseProcessDetails[_]): String =
+    text
+      .replace("{{processName}}", process.name)
```

Comment:
why we use react-style on backend?

Reply:
What's your proposition for templating?

Reply:
It's not only react-style.. ;)

Reply:
on backend configuration we use more $key style...

Reply:
Changed.